### PR TITLE
Support no sharding config

### PR DESCRIPTION
**Pull request by zhaojuanmao**
supporting no sharding config to make it similar to DDP algorithm

### CODE REVIEW

File: torch/distributed/fsdp/fully_sharded_data_parallel.py
Comment (mrshenli) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:107]: nit: same as on the previous PR. Double quote `DistributedDataParallel` and `all_reduce` to display as code.

**Review by mrshenli — APPROVED** [review]
Reviewed on #76628, stamp to unblock. 

Please remove changes in `ideep` submodule before landing.

**Comment by zhaojuanmao** [comment]
> Reviewed on #76628, stamp to unblock.
> 
> Please remove changes in `ideep` submodule before landing.

how to remove changes in `ideep` submodule..

**Comment by zhaojuanmao** [comment]
> > Reviewed on #76628, stamp to unblock.
> > Please remove changes in `ideep` submodule before landing.
> 
> how to remove changes in `ideep` submodule..

removed

**Comment by zhaojuanmao** [comment]
the commits messed up:(, closing this, will create a new PR

**Pull request closed without merging**
